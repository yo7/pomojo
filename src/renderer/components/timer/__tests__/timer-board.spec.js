import Vue from 'vue'
import Vuex from 'vuex'
import {createRenderer} from 'vue-server-renderer'
import {shallow} from 'vue-test-utils'
import TimerBoard from '../timer-board.vue'

Vue.use(Vuex)

describe('TimerBoard', () => {
  const state = {
    timer: {
      seconds: 1500,
      running: false,
      resting: false,
      pausing: true,
      today: 0
    },
    preferences: {
      goal: 8
    }
  }
  const store = new Vuex.Store({state})
  const wrapper = shallow(TimerBoard, {store})

  it('renders correctly', () => {
    const renderer = createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error()
      }
      expect(str).toMatchSnapshot()
    })
  })
})
