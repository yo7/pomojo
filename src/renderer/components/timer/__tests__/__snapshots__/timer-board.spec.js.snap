// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerBoard renders correctly 1`] = `
<div class="timer-board">
    <!---->
    <!---->
    <!---->
    <!---->
</div>
`;
